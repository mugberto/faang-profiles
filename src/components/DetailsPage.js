import Grid from '@material-ui/core/Grid';
import { makeStyles } from '@material-ui/core';
import StockChart from './StockChart';

const useStyle = makeStyles({
  hero: {
    padding: '3em',
    backgroundColor: '#FB5294',
  },
  icon: {
    display: 'inline-block',
    borderRadius: '50%',
    border: '2px solid white',
    height: '20px',
  },
  boxlight: {
    backgroundColor: '#DB4681',
    padding: '1em',
    height: '10rem',
  },
  boxDark: {
    backgroundColor: '#D1447A',
    padding: '1em',
    height: '10rem',
  },
  boxTitle: {
    backgroundColor: '#DF4783',
    fontSize: '0.85rem',
    padding: '0.5em',
  },
  companyName: {
    fontWeight: '900',
    textTransform: 'uppercase',
  },
  logo: {
    width: '65px',
    objectFit: 'contain',
    filter: 'grayscale(100%)',
    mixBlendMode: 'multiply',
  },
  boxLink: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'flex-end',
    justifyContent: 'space-between',
    height: '100%',
    textDecoration: 'none',
  },
  chartBG: {
    backgroundColor: '#E84987',
    padding: '1rem',
  },
});

const DetailsPage = () => {
  const firmDetails = {
    symbol: 'AAPL',
    price: 113.02,
    beta: 1.34434,
    volAvg: 172070917,
    mktCap: 1932924420000,
    lastDiv: 0.7825,
    range: '53.1525-137.98',
    changes: -3.77,
    companyName: 'Apple Inc',
    currency: 'USD',
    isin: 'US0378331005',
    cusip: '037833100',
    exchange: 'Nasdaq Global Select',
    exchangeShortName: 'NASDAQ',
    industry: 'Consumer Electronics',
    website: 'https://www.apple.com/',
    description: "Apple, Inc. engages in the design, manufacture, and sale of smartphones, personal computers, tablets, wearables and accessories, and other variety of related services. The company is headquartered in Cupertino, California and currently employs 137,000 full-time employees. The company is considered one of the Big Four technology companies, alongside Amazon, Google, and Microsoft. The firm's hardware products include the iPhone smartphone, the iPad tablet computer, the Mac personal computer, the iPod portable media player, the Apple Watch smartwatch, the Apple TV digital media player, the AirPods wireless earbuds and the HomePod smart speaker. Apple's software includes the macOS, iOS, iPadOS, watchOS, and tvOS operating systems, the iTunes media player, the Safari web browser, the Shazam acoustic fingerprint utility, and the iLife and iWork creativity and productivity suites, as well as professional applications like Final Cut Pro, Logic Pro, and Xcode. Its online services include the iTunes Store, the iOS App Store, Mac App Store, Apple Music, Apple TV+, iMessage, and iCloud. Other services include Apple Store, Genius Bar, AppleCare, Apple Pay, Apple Pay Cash, and Apple Card.",
    ceo: 'Mr. Timothy Cook',
    sector: 'Technology',
    country: 'US',
    fullTimeEmployees: '137000',
    phone: '14089961010',
    address: '1 Apple Park Way',
    city: 'Cupertino',
    state: 'CALIFORNIA',
    zip: '95014',
    dcfDiff: 89.92,
    dcf: 123.527,
    image: 'https://financialmodelingprep.com/image-stock/AAPL.jpg',
    ipoDate: '1980-12-12',
  };

  const classes = useStyle();

  return (
    <Grid container spacing={0}>
      <Grid item xs={12} container className={classes.hero}>
        <Grid item xs={6} container justifyContent="flex-end" style={{ paddingRight: '8px' }}>
          <img src={firmDetails.image} alt="logo" className={classes.logo} />
        </Grid>
        <Grid item xs={6} container direction="column" justifyContent="center" alignItems="start" style={{ paddingLeft: '8px' }}>
          <h1 style={{ lineHeight: '1', fontSize: '1.2rem' }}>
            {firmDetails.companyName}
          </h1>
          <p>
            {firmDetails.address}
            {' '}
            <br />
            {firmDetails.city}
            {' '}
            <br />
            {firmDetails.state}
            {' '}
            <br />
            {firmDetails.phone}
          </p>
        </Grid>
      </Grid>
      <Grid item xs={12} className={classes.boxTitle}>STOCK CHART</Grid>
      <Grid item xs={12} className={classes.chartBG}>
        <StockChart />
      </Grid>
      <Grid item xs={12} className={classes.boxTitle}>DESCRIPTION</Grid>
      <Grid item xs={12} className={classes.chartBG}>
        <p>{firmDetails.description}</p>
      </Grid>
    </Grid>
  );
};

export default DetailsPage;
