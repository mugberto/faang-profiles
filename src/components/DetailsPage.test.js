import React from 'react';
import { render, screen } from '../test-utils';
import DetailsPage from './DetailsPage';

jest.mock('react-router-dom', () => ({
  useParams: () => ({ symbol: 'AAPL' }),
}));

describe('DetailsPage', () => {
  it('renders page correctly', () => {
    const tree = render(<DetailsPage />);
    expect(tree).toMatchSnapshot();
  });

  it('updates the UI with loaded data is loaded', async () => {
    render(<DetailsPage />);
    expect(await screen.findByText('STOCK CHART')).toBeInTheDocument();
  });
});
