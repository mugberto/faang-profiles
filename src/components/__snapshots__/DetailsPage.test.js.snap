// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsPage renders page correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiGrid-root MuiGrid-container"
      >
        <div
          class="MuiGrid-root makeStyles-hero-1 MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-content-xs-flex-end MuiGrid-grid-xs-6"
            style="padding-right: 8px;"
          />
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-justify-content-xs-center MuiGrid-grid-xs-6"
            style="padding-left: 8px;"
          >
            <h1
              style="line-height: 1; font-size: 1.2rem;"
            />
            <p>
              
               
              <br />
              
               
              <br />
              
               
              <br />
              
            </p>
          </div>
        </div>
        <div
          class="MuiGrid-root makeStyles-boxTitle-5 MuiGrid-item MuiGrid-grid-xs-12"
        >
          STOCK CHART
        </div>
        <div
          class="MuiGrid-root makeStyles-chartBG-9 MuiGrid-item MuiGrid-grid-xs-12"
        />
        <div
          class="MuiGrid-root makeStyles-boxTitle-5 MuiGrid-item MuiGrid-grid-xs-12"
        >
          DESCRIPTION
        </div>
        <div
          class="MuiGrid-root makeStyles-chartBG-9 MuiGrid-item MuiGrid-grid-xs-12"
        >
          <p />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiGrid-root MuiGrid-container"
    >
      <div
        class="MuiGrid-root makeStyles-hero-1 MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-content-xs-flex-end MuiGrid-grid-xs-6"
          style="padding-right: 8px;"
        />
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-justify-content-xs-center MuiGrid-grid-xs-6"
          style="padding-left: 8px;"
        >
          <h1
            style="line-height: 1; font-size: 1.2rem;"
          />
          <p>
            
             
            <br />
            
             
            <br />
            
             
            <br />
            
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root makeStyles-boxTitle-5 MuiGrid-item MuiGrid-grid-xs-12"
      >
        STOCK CHART
      </div>
      <div
        class="MuiGrid-root makeStyles-chartBG-9 MuiGrid-item MuiGrid-grid-xs-12"
      />
      <div
        class="MuiGrid-root makeStyles-boxTitle-5 MuiGrid-item MuiGrid-grid-xs-12"
      >
        DESCRIPTION
      </div>
      <div
        class="MuiGrid-root makeStyles-chartBG-9 MuiGrid-item MuiGrid-grid-xs-12"
      >
        <p />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
